views/default/user.html

```diff
@@ -221,10 +221,14 @@ <h2>Runestone Registration</h2>
                         <script type="text/javascript">
                             function checkTcp() {
                                 let cb = document.getElementById("auth_user_accept_tcp");
-                                if (cb.checked) {
+
+                                var reCaptchaKey = grecaptcha.getResponse();
```

Comment:
This does not look right to me.  You care calling the `getReponse` method two times in two lines of code.  If that is really how it is supposed to work please ad a comment to explain why this is correct.

This also adds the reCaptcha functionality unconditionally?  I would prefer that it remains an option.

Reply:
We're not quite sure what you mean in your first comment. The way that it works currently is that it checks to see if there is a token issued by reCaptcha, and if there is, then it passes and the user is allowed to go on. Basically reCaptcha is a true or false sort of deal, the user passes or fails. 

As for the unconditional part, the option to toggle reCaptcha on or off is still in 1.py. If it's set to true, reCaptcha will be enabled on the page. If it's false, it won't be. I checked to make sure in our build environment if having it disabled and going through would cause any problems, but it seems to work as intended.

Reply:
The way it is written is super confusing, unless you are very familiar with how the captcha code works.

lines 225 and 226:

```
var reCaptchaKey = grecaptcha.getResponse();
let reCaptchaValidation = Boolean(grecaptcha.getResponse());
```

Line 225 looks like you are getting a key and the next line looks like you are calling get response to get something that is a boolean.  To my eye this looks like a mistake.  If the intention is to simply see if you received a key then you should document that or just use `Boolean(reCaptchaKey)` or `reCaptchaKey !=== ""`